### PR TITLE
 Fix unlock temp dir label mounted over /usr 

### PR DESCRIPTION
**Pull request by rhatdan**
When an admin executes `ostree admin unlock`, the command creates
an overlay director in /var/tmp that is later mounted on /usr.  The
problem on an SELinux system you will end up with /usr being labeled
system_u:object_r:tmp_t:s0, when the original label was
system_u:object_r:usr_t:s0.  This is causing errors with atomic scan
on unlocked systems.  And could potentially cause other SELinux issues.

### CODE REVIEW

**Comment by rhatdan** [comment]
@miabbott PTAL

**Comment by rhatdan** [comment]
Should fix https://github.com/projectatomic/atomic-host-tests/pull/119

**Comment by rhatdan** [comment]
@miabbott I have not tested but this should fix the issue you are seeing.

**Comment by cgwalters** [comment]
In ostree we have an (IMO) nice wrapper for libselinux context stuff; I pushed a rewritten version to use it above ⬆️.  Testing now.

**Comment by cgwalters** [comment]
Ooh, in testing `self->sepolicy` was a trap.  This new version works for me.

@rh-atomic-bot delegate=rhatdan

File: src/libostree/ostree-sysroot.c [hunk 1]
```diff
@@ -1765,8 +1771,16 @@ ostree_sysroot_deployment_unlock (OstreeSysroot     *self,
...
+        /* Ensure that the directory is created with the same label as `/usr` */
+        { g_auto(OstreeSepolicyFsCreatecon) con = { 0, };
+
+          if (!_ostree_sepolicy_preparefscreatecon (&con, sepolicy,
+                                                    "/usr", 0755, error))
+            goto out;
+
+          if (!glnx_mkdtempat (AT_FDCWD, development_ovldir, 0755, error))
+            goto out;
+        }
```
Comment (rhatdan) [inline src/libostree/ostree-sysroot.c]: Should you set the fscreatecon back to default, in csae ostree creates any other files/directories?

Comment (cgwalters) [inline src/libostree/ostree-sysroot.c]: That's exactly what the ` { g_auto(OstreeSepolicyFsCreatecon) con = { 0, };` bit does - we make extensive use of the [GCC cleanup attribute](https://blog.verbum.org/2012/05/09/__attribute__-cleanup-or-how-i-came-to-love-c-again/).

Comment (rhatdan) [inline src/libostree/ostree-sysroot.c]: Excellent, Thanks.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+ ab1ac9d

**Comment by miabbott** [comment]
Worked for me! 

See details here:  https://github.com/projectatomic/atomic/issues/951#issuecomment-289796585

**Pull request closed without merging**